dev/dev-env-setup.sh

```diff
@@ -0,0 +1,232 @@
+
+  dev/dev-env-setup.sh -d $(pwd)/.venvs/mlflow-dev
+
+  Note: it is recommended to preface virtualenv locations with a directory name prefaced by '.' (i.e., ".venvs").
+
+  The default environment setup is for basic functionality, installing the minimum development requirements dependencies.
+  To install the full development environment that supports working on all flavors and running all tests locally, set
+  the flag '-f' or '--full'
+
+-h, -help,        --help        Display help
+
+-d, -directory    --directory   The path to install the virtual environment into
+
+-f, -full         --full        Whether to install all dev requirements (Default: false)
+
+-v, -verbose      --verbose     Whether to fill stdout with every command or not
+
+-q, -quiet        --quiet       Whether to have pip install in quiet mode (Default: false)
+
+EOF
+}
+
+export verbose=0
+export quiet=0
+while getopts "d:fvqh" opt
+do
+  case "$opt" in
+    d) directory="$OPTARG" ;;
+    f) full_install=1 ;;
+    v) verbose=1 ;;
+    q) quiet=1 ;;
+    h) showHelp; exit ;;
+    *) showHelp; exit ;;
+  esac
+done
+
+if [[ $verbose == 1 ]]; then
+  set -exv
+fi
+
+# Acquire the OS for this environment
+case "$(uname -s)" in
+  Darwin*)                       machine=mac;;
+  Linux*)                        machine=linux;;
+  *)                             machine=unknown;;
+esac
+
+function quietpip(){
+  echo $( (( quiet == 1 && verbose == 0 )) && printf %s '-q' )
+}
+
+# Check if pyenv is installed and offer to install it if not present
+pyenv_exist=$(command -v pyenv)
+
+if [ -z "$pyenv_exist" ]; then
+  if [ -z "$GITHUB_ACTIONS" ]; then
+    read -p "pyenv is required to be installed to manage python versions. Would you like to install it? $(tput bold)(y/n)$(tput sgr0): " -n 1 -r
+    echo
+  fi
+  if [[ $REPLY =~ ^[Yy]$ || -n "$GITHUB_ACTIONS" ]]; then
+    if [[ "$machine" == mac ]]; then
+      # Check if brew is installed and install it if it isn't present
+      # Note: if xcode isn't installed, this will fail.
+      if [ -z "$(command -v brew)" ]; then
+        echo "Brew is required to install pyenv on MacOS. Installing in your home directory."
+        bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
+      fi
+      echo "Updating brew and installing pyenv..."
+      echo "Note: this will probably take a considerable amount of time."
+      brew update
+      brew install pyenv
+      brew install openssl readline sqlite3 xz zlib
+    elif [[ "$machine" == linux ]]; then
+      sudo apt-get update -y
+      sudo apt-get install -y make build-essential libssl-dev zlib1g-dev \
+      libbz2-dev libreadline-dev libsqlite3-dev wget curl llvm \
+      libncursesw5-dev xz-utils tk-dev libxml2-dev libxmlsec1-dev libffi-dev liblzma-dev
+      # Install pyenv from source
+      git clone --depth 1 https://github.com/pyenv/pyenv.git "$HOME/.pyenv"
+      PYENV_ROOT="$HOME/.pyenv"
+      PYENV_BIN="$PYENV_ROOT/bin"
```

Comment:
```suggestion
      PYENV_ROOT="$HOME/.pyenv"
      PYENV_BIN="$PYENV_ROOT/bin"
      PATH="$PYENV_BIN:$PATH"
```

I think we need to add `PYENV_BIN` to `PATH`.